packages/core/src/bundle.ts

```diff
@@ -11,10 +13,21 @@ import { Bundle } from '@medplum/fhirtypes';
  * @returns transaction type bundle
  */
 export function convertToTransactionBundle(bundle: Bundle): Bundle {
+  const idToUuid: Record<string, string> = {};
+  bundle = deepClone(bundle);
   for (const entry of bundle.entry || []) {
     delete entry.resource?.meta;
-    entry.fullUrl = 'urn:uuid:' + entry.resource?.id;
-    delete entry.resource?.id;
+    const id = entry.resource?.id;
+    // In some FHIR example bundles,they use human-readable ids, rather than true uuids.
+    if (id) {
+      if (isUUID(id)) {
+        idToUuid[id] = id;
+      } else {
+        idToUuid[id] = crypto.randomUUID();
```

Comment:
This works in browser environments, but not in NodeJS environments.  

We have a cross-environment version in `fhir-router`: https://github.com/medplum/medplum/blob/main/packages/fhir-router/src/repo.ts#L371

We could move that into `core` if we need to.